lib/modules/datasource/docker/index.ts

```diff
@@ -464,6 +465,41 @@ export class DockerDatasource extends Datasource {
     }
   }
 
+  @cache({
+    namespace: 'datasource-docker-imageconfig',
+    key: (
+      registryHost: string,
+      dockerRepository: string,
+      configDigest: string
+    ) => `${registryHost}:${dockerRepository}@${configDigest}`,
+    ttlMinutes: 1440 * 28,
+  })
+  public async getImageConfig(
+    registryHost: string,
+    dockerRepository: string,
+    configDigest: string
+  ): Promise<HttpResponse<ImageConfig> | undefined> {
+    logger.debug(
+      `getImageConfig(${registryHost}, ${dockerRepository}, ${configDigest})`
+    );
+
+    const headers = await getAuthHeaders(
+      this.http,
+      registryHost,
+      dockerRepository
+    );
+    // istanbul ignore if: Should never be happen
+    if (!headers) {
+      logger.debug('No docker auth found - returning');
```

Comment:
```suggestion
      logger.warn('No docker auth found - returning');
```

 @rarkins I think we should warn, as this case should never happen?

Reply:
If a registry specifies an auth challenge in `WWW-Authenticate` that renovate doesn't understand, `headers` will be null. 

Note that the same `logger.debug` stmt is also used in `getManifestResponse()` and `getLabels()`. 

https://github.com/renovatebot/renovate/blob/930efb5896eba1cfa032ca7872ef6cb7c25b00ab/lib/modules/datasource/docker/index.ts#L402

https://github.com/renovatebot/renovate/blob/930efb5896eba1cfa032ca7872ef6cb7c25b00ab/lib/modules/datasource/docker/index.ts#L695

Reply:
Maybe it can be good to warn so that these problems can be surfaced?